homeassistant/components/homekit/type_covers.py

```diff
@@ -20,13 +19,10 @@ class WindowCovering(HomeAccessory):
     The cover entity must support: set_cover_position.
     """
 
-    def __init__(self, hass, entity_id, display_name, **kwargs):
+    def __init__(self, hass, name, entity_id, **kwargs):
         """Initialize a WindowCovering accessory object."""
-        super().__init__(display_name, entity_id,
-                         CATEGORY_WINDOW_COVERING, **kwargs)
-
-        self.hass = hass
-        self.entity_id = entity_id
+        super().__init__(hass, name, entity_id,
+                         category=CATEGORY_WINDOW_COVERING, **kwargs)
```

Comment:
Personally I would really like to move these calls to the `HomeAccessory` class somehow. Any ideas who to improve that?

Reply:
I don't like the decorator and not sure how that will interfere with `__init__`

You could do this (note: `category=` at the end so they look the same at first glance)

```python
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs, category=CATEGORY_LIGHT)
```

but being explicit might give you some code editor hints, so I like this version more:

```python
    def __init__(self, hass, name, entity_id, **kwargs):
        super().__init__(hass, name, entity_id, **kwargs,
                         category=CATEGORY_LIGHT)
```

Reply:
The idea behind the decorator was to limit code redundancy between different types. You're right the it might be more difficult for editors to read, however I don't think that is that big of a deal since those attributes are already assigned in `accessories.py/HomeAccessory` and most inexperience developers will only copy existing types without fully understanding them.

Regarding the interference with `__init__`, it should change anything. The call order is as follows:
 1. HA start
 2. Module is loaded and `__init__` call is replaced by decorater
 3. Call to `light.__init__`
 4. `super().__init__()`
 5. `__init__()`

Reply:
If you dont like the super rather add a bootstrap/postinit method in the child classes that you call from the parent’s init. Type could be a class var on the child

Reply:
In HomeAccessory you then have:

```python
    def __init__(self, hass, name, entity_id, **kwargs):
        """Initialize a Accessory object."""
        super().__init__(name, **kwargs)
        set_accessory_info(self, name, model=entity_id)
        self.entity_id = entity_id
        self.hass = hass
        cat = self.init_category()
        self.category = getattr(Category, cat, Category.OTHER)
```

with only `def init_category(self)` and no more `__init__` in the rest, init_category then returns the category.

Reply:
Extending on your idea what do you think about the following:
 - Making the `category` a static var of the `type`
 - Changing the `__init__()` of the types to something like `init_setup()` which is called from `HomeAccessory.__init__()`

Reply:
`init_setup()` might not be the best name. If you have something else in mind, I'm happy to change it.
The only caveat with this solution is, that I had to add `# pylint: disable=attribute-defined-outside-init` to every `type` file.